### DISCUSSION TITLE
Unable to use print functionality with StyledText

### DISCUSSION THREAD

**Discussion by Github-Satya-BMC** [eclipse-platform/eclipse.platform.swt#493] [General]
We are overriding org.eclipse.jface.action.Action for a PrintAction to print my custom editor text in the detailed style manner(margin, line numbering, text coloring and styling etc).
This was working perfectly with earlier versions(Version: Photon release (4.8.0)) of eclipse, however in later version(Version: 2020-09 (4.17.0)) we are getting an exception shared below:
		**Caused by: java.lang.IllegalArgumentException: Argument not valid
		at org.eclipse.swt.SWT.error(SWT.java:4704)
		at org.eclipse.swt.SWT.error(SWT.java:4638)
		at org.eclipse.swt.SWT.error(SWT.java:4609)
		at org.eclipse.swt.custom.StyledText.setStyleRanges(StyledText.java:10229)
	        at org.eclipse.swt.custom.StyledText.setStyleRanges(StyledText.java:10444)**

My question here is:
	Is there any changes to Version: 2020-09 (4.17.0) with regards to 
	setStyleRanges(StyleRange[] ranges) method ?

**Comment by mickaelistria** [comment]
> Is there any changes to Version: 2020-09 (4.17.0) with regards to setStyleRanges(StyleRange[] ranges) method

That's something you can and should check yourself.

**Reply by Github-Satya-BMC** [reply]
I have checked it corresponding to my problem , but didn't able to find something significant. Posted this issue and hopefully get something so that I can proceed a bit.
